docs/remote_debugging.md

```diff
@@ -0,0 +1,218 @@
+DEBUGGER: Session start (pid: 7773)
+DEBUGGER: Debugger can attach via UNIX domain socket (/home/ko1/.ruby-debug-sock/ruby-debug-ko1-7773)
+DEBUGGER: wait for debugger connection...
+```
+
+By default, `rdbg --open` uses UNIX domain socket and generates path name automatically (`/home/ko1/.ruby-debug-sock/ruby-debug-ko1-7773` in this case).
+
+## `require 'debug/open'`
+
+If you can modify the program, you can open debugging port by adding `require 'debug/open'` to the program.
+
+If you don't want to stop the program at the beginning, you can also use `require 'debug/open_nonstop'`.
+
+**Please do not leave these requires in your program as they will allow other people to connect to it.**
+
+## `DEBUGGER__.open`
+
+After requiring `debug/session`, you can start debug session with the following methods.
+They are convenient if you want to specify debug configurations with Ruby code.
+
+* `DEBUGGER__.open(**kw)`: open debug port with configuration (without configurations open with UNIX domain socket)
+* `DEBUGGER__.open_unix(**kw)`: open debug port with UNIX domain socket
+* `DEBUGGER__.open_tcp(**kw)`: open debug port with TCP/IP
+
+For example:
+
+```ruby
+require 'debug/session'
+DEBUGGER__.open
+
+# your application code
+```
+
+## TCP/IP
+
+If you want to use TCP/IP for remote debugging, you need to specify the port and host with `--port` like
+
+```
+rdbg --open --port 12345 # binds to 127.0.0.1:12345
+rdbg --open --port 12345 --host myhost.dev # binds to myhost.dev:12345
+```
+
+If you don't use the `rdbg` executable, you can also specify port and host with `RUBY_DEBUG_PORT` and `RUBY_DEBUG_HOST` environment variables:
+
+```shell
+$ RUBY_DEBUG_PORT=12345 RUBY_DEBUG_HOST=localhost ruby target.rb
+```
+
+# Connect To A Remote Debuggee
+
+## Debugger Console
+
+You can connect to the debuggee with `rdbg --attach` command (`rdbg -A` for short).
+
+```shell
+$ rdbg --attach
+[1, 7] in target.rb
+=>    1| a = 1
+      2| b = 2
+      3| c = 3
+      4| d = 4
+      5| p [a, b, c, d]
+      6|
+      7| __END__
+=>#0    <main> at target.rb:1
+
+(rdbg:remote)
+```
+
+If there is only one opened UNIX domain sockets in the socket directory, `rdbg --attach` will connect to it automatically.
+
+If there are more than one socket files, you need to specify the socket name after the `--attach` flag:
+
+```
+❯ rdbg --attach
+Please select a debug session:
+  ruby-debug-st0012-64507
+  ruby-debug-st0012-68793
+
+❯ rdbg --attach ruby-debug-st0012-64507
+[1, 1] in target.rb
+=>   1| a = 1
+=>#0    <main> at target.rb:1
+(rdbg:remote)
+```
+
+When `rdbg --attach` connects to the debuggee, you can use any debugger commands like in local debugger console. When a debuggee program exits, the remote console will also terminate.
+
+NOTE: If you use the `quit` command, only the remote console exits and the debuggee program continues to run (and you can connect to it again). If you want to exit the debuggee program as well, use `kill` command instead.
```

Comment:
Maybe we want to use this fancy new GitHub Markdown feature 😄 

Preview:

<img width="972" alt="Screen Shot 2022-07-04 at 5 07 53 PM" src="https://user-images.githubusercontent.com/15860699/177216608-36fc590e-906a-4955-8883-74729b41ce70.png">


```suggestion
> **Note**
> If you use the `quit` command, only the remote console exits and the debuggee program continues to run (and you can connect to it again). If you want to exit the debuggee program as well, use the `kill` command instead.
```